docs/content/release-process.md

```diff
@@ -13,11 +13,23 @@ To cut a release of Spin, you will need to do the following:
    - Bump the version in Spin's `Cargo.toml`
    - Update SDK_VERSION in `templates/Makefile`
    - Check the docs for hard-coded version strings
+   - For a major release (e.g. `0.9.0` to `0.10.0`), update the versions
+     in sdk/rust/src/lib.rs and sdk/go/sdk-version-go.c
+      - e.g. `spin-sdk-version-0-10-pre0` becomes `spin-sdk-version-0-10`
+      - See [sips/011-component-versioning.md](sips/011-component-versioning.md)
+        for details
 1. Merge the PR created in #1 (Such PRs are still required to get approvals, so
    make sure you get signoff on the PR)
 1. Before proceeding, verify that the merge commit on `main` intended to be
    tagged is green, i.e. CI is successful
 1. Create a new tag with a `v` and then the version number (`v0.9.1`)
+1. For a major release (e.g. `0.9.0` to `0.10.0`), update the versions in
```

Comment:
Appreciate adding these steps here until we have them automated.

We could potentially update the `spin-sdk-version-0-X(-pre0)` string in both SDKs via a few make targets, as we do for the [templates](https://github.com/fermyon/spin/blob/main/templates/Makefile).  (Corresponding automation [here](https://github.com/fermyon/spin/blob/main/.github/workflows/release.yml#L234-L271).)

Whichever approach we choose, perhaps we can track in the same ticket for adding `spin-sdk-commit-$HASH`...

Reply:
@vdice This is subtly different, though, I think, because it updates the strings to the _next_ anticipated release, rather than the _current_ release, so that canaries built _after_ the release provide informative failures.

Reply:
@vdice Sorry - I think I might have misread your comment.  Please ignore me!

To be clear, I would _love_ to see an automated roll forward after a release, and I'd like to see it applied to the app version in Cargo.toml as well.  This was always intended so that template tagging would work correctly with canaries, but never quite happened.

Reply:
@itowlson ah, the templates would also benefit from similar post-release roll forward automation?  If so, it'd seem natural to scope that in with the same automation work.

Reply:
I went ahead and automated this a bit, so now the only extra manual step is to update the version strings in `main` to `$MAJOR.$MINOR.$PATCH-pre0` after tagging the release.  Which we should definitely automate at some point.